src/types.ts

```diff
@@ -29,20 +29,31 @@ export type SchemaType =
   | "[]int"
   | "[]uint"
   | "[]uuid";
-// TODO: index signature is a better fit here imo.
-// also look into eslint config to allow for usage of index signatures
-export type Schema = Record<
-  string,
-  {
-    type?: SchemaType;
-    filterable?: boolean;
-    /**
-     * @deprecated use `full_text_search` instead
-     */
-    bm25?: boolean | Partial<FTSParams>;
-    full_text_search?: boolean | Partial<FTSParams>;
-  }
->;
+
+export type Schema =
+  | {
+      [K in string]: {
```

Comment:
this `[K in string]` definition seems like non-standard index signature syntax, is there a reason you didn't use `[key: string]`?

Reply:
eslint shouts `A record is preferred over an index signature.` and doing this silenced it! I can definitely silence it by adding a comment if you think that's better?

Reply:
yes, we shouldn't ever make code harder to read just because it makes the linter happy